js/interactive-guides/modules/tabbed-editor.js

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+var tabbedEditor = (function() {
+    var __editors = [];
+
+    var tabbedEditorType = function(container, stepName, content) {
+        var deferred = new $.Deferred();
+
+        this.stepName = stepName;
+        this.editorList = content.editorList || [];
+
+        __loadAndCreate(this, container, stepName, content).done(function(result){
+            deferred.resolve(result);
+        });
+        return deferred;
+    };
+    
+    tabbedEditorType.prototype = {
+
+        addEditor: function(editorInfo) {
+            // New tabs will be added to the end of the existing tabs
+            var numTabs = this.$teTabList.find('a').length;
+
+            // A fileName must be specified for each editor.
+            // Editor preload, readonly, save, and callback fields are optional.
+            // See editor.js for more information.
+            if (!editorInfo.fileName) {
+                editorInfo.fileName = "Tab-" + numTabs;
+            }
+
+            // Create the dom elements for the new editor
+            var editorName = 'teTab-editor' + this.displayTypeNum + '-tab' + numTabs;    // or should this be the filename?
```

Comment:
What you have is fine as long as the editorName is unique. Using fileName makes it easier to associate the id with the file it is for but fileName could be long.